libgit/browser.go

```diff
@@ -39,6 +40,12 @@ type Browser struct {
 	tree  *object.Tree
 	root  string
 	mtime time.Time
+
+	lock sync.RWMutex
+	// Caches ReadDir results. Since the *Browser object gets blown away when
+	// HEAD changes, we don't need to worry about invalidation -- unless memory
+	// footprint becomes a problem.
+	dirChildrenCache map[string][]os.FileInfo
```

Comment:
Also, it's a shame this doesn't help improve `LStat/Stat` commands.  Can we give it another level indexed by entry name, instead of having it as an array?  Or would reconstructing the array from that be too expensive?

Reply:
I'm thinking we should probably plumb through a global `*lru.Cache` to be shared among all `*Browser` instances. For `kbpagesd` I think this would make a big difference for more active sites that are over the capacity, compared to having per-browser caches. WDYT?

Reply:
> Can we give it another level indexed by entry name, instead of having it as an array? Or would reconstructing the array from that be too expensive?

Good idea; I think it should be fine, but will do some simple benchmark to be sure.